runtime/reconcilers/alert.go

```diff
@@ -561,30 +584,40 @@ func (r *AlertReconciler) executeSingleWrapped(ctx context.Context, self *runtim
 		queryForAttrs = a.Spec.GetQueryForAttributes().AsMap()
 	}
 
-	// Create and execute query
-	q, err := queries.ProtoToQuery(qpb, queryForAttrs)
-	if err != nil {
-		return nil, fmt.Errorf("failed to build query: %w", err)
-	}
-	err = r.C.Runtime.Query(ctx, r.C.InstanceID, q, alertQueryPriority)
+	res, err := r.C.Runtime.Resolve(ctx, &runtime.ResolveOptions{
+		InstanceID: r.C.InstanceID,
+		Resolver:   "legacy_metrics",
+		ResolverProperties: map[string]any{
+			"query_name":      a.Spec.QueryName,
+			"query_args_json": a.Spec.QueryArgsJson,
+			"is_formatted":    true,
+			"row_count":       1,
+		},
```

Comment:
How about moving this logic into `parse_alert.go` and completely removing `Spec.QueryName` and `Spec.QueryArgsJSON` from the spec? I.e. if the YAML has a `Query`, it just sets the spec's resolver to to `legacy_metrics` right away?

Each time the runtime is restarted, it re-parses all YAML files and overwrites any resources that have been changed, so this should be safe (there might be a small race condition on the restart around whether an alert is checked before the re-parse is complete – so just need to maintain the `if a.Spec.Resolver != "" {` in here to avoid crashes).